src/app/_components/banner.tsx

```diff
@@ -0,0 +1,22 @@
+import { BannerBody } from "@/app/_components/banner-body";
+import { getAllPosts } from "@/lib/api";
+
+export async function Banner() {
+  const bannerPost = getAllPosts().find((post) => post.banner);
+
+  if (
+    !bannerPost ||
+    !bannerPost.banner || // redundant check to satisfy type checking
+    bannerPost.banner.hide === true
```

Comment:
See if this will make it happy:

```suggestion
    bannerPost?.banner?.hide === true
```

Reply:
Or change the find above to be `(post) => post.banner && !post.banner.hide)

Reply:
Still need to return `null` if `bannerPost` is not found. You can do `banner?` but then you need a `banner!` on the `bannerData` declaration or put a lot more question marks everywhere else.

Reply:
Eh I suppose this case is what ! is meant for. I just remember reading it was frowned upon somewhere